mautic/mautic
# Sparkpost error handling update

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | Y

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
[Sparkpost is removing](https://www.sparkpost.com/blog/error-handling-transmissions-api/) `description` property from the error responses. This PR addresses that. And while writing tests for the `send()` method I had to use the SparkpostFactory that was used for Momentum, update it for Sparkpost to use the default values and make some other changes to make the tests pass. I think some of them are bugs.

Another change is that Sparkpost won't control email template validation on send. This PR implements pre-check to validate email template. The send will not happen if the template is invalid.

#### Steps to test this PR:
1. Make sure you use the Sparkpost email transport
2. Send individual and segment email with some tokens in it.
3. Ensure that the emails were delivered properly.
